client/data/media/use-delete-media-mutation.js

```diff
@@ -2,8 +2,9 @@ import { useMutation } from '@tanstack/react-query';
 import wp from 'calypso/lib/wp';
 
 export const useDeleteMediaMutation = ( queryOptions = {} ) => {
-	return useMutation(
-		( { siteId, mediaId } ) => wp.req.post( `/sites/${ siteId }/media/${ mediaId }/delete` ),
-		queryOptions
-	);
+	return useMutation( {
+		mutationFn: ( { siteId, mediaId } ) =>
+			wp.req.post( `/sites/${ siteId }/media/${ mediaId }/delete` ),
+		...queryOptions,
```

Comment:
The subtle downside with this approach is that if `queryOptions` has `mutationFn` (although it shouldn't), it will override the one we specify above. Perhaps we should reorder them and have `mutationFn` declared below `...queryOptions`? WDYT?

Reply:
Agreed, I had the exact same thing in mind when refactoring those. Overriding the `mutationFn` would kinda defeat the purpose of having a custom mutation hook in the first place (so it's not likely). One possible solution I can see (other than switching the arguments around) is to destructure only what we expect from the `queryOptions` objects. Another thing which comes to mind is typing `queryOptions`.